spyder/plugins/ipythonconsole/widgets/tests/test_kernelconnect.py

```diff
@@ -0,0 +1,220 @@
+            keyring.set_password("existing_kernel",
+                                 "ssh_key_passphrase", "")
+            keyring.set_password("existing_kernel",
+                                 "ssh_password", "")
+        except Exception:
+            pass
+
+    # test form values
+    pytest.cf_path = "cf_path"
+    pytest.un = "test_username"
+    pytest.hn = "test_hostname"
+    pytest.pn = 123
+    pytest.kf = "test_kf"
+    pytest.kfp = "test_kfp"
+    pytest.pw = "test_pw"
+
+    request.addfinalizer(teardown)
+    return DialogFactory()
+
+
+# =============================================================================
+# Tests
+# =============================================================================
+def test_kernel_connection_dialog_remember_input_ssh_passphrase(
+        qtbot, connection_dialog_factory):
+    """
+    Test that the dialog remember user's kernel connection
+    settings and ssh key passphrase when the user checks the
+    save checkbox.
+    """
+
+    dlg = connection_dialog_factory.submit_filled_dialog(use_keyfile=True,
+                                                         save_settings=True)
+
+    # Press ok and save connection settings
+    qtbot.mouseClick(dlg.accept_btns.button(QDialogButtonBox.Ok),
+                     Qt.LeftButton)
+
+    # create new dialog and check fields
+    new_dlg = connection_dialog_factory.get_default_dialog()
+    assert new_dlg.cf.text() == pytest.cf_path
+    assert new_dlg.rm_group.isChecked()
+    assert new_dlg.hn.text() == pytest.hn
+    assert new_dlg.un.text() == pytest.un
+    assert new_dlg.pn.text() == str(pytest.pn)
+    assert new_dlg.kf.text() == pytest.kf
+    if (not sys.platform.startswith('linux') or
+            not os.environ.get('CI') is not None):
+        assert new_dlg.kfp.text() == pytest.kfp
+
+
+def test_kernel_connection_dialog_doesnt_remember_input_ssh_passphrase(
+        qtbot, connection_dialog_factory):
+    """
+    Test that the dialog doesn't remember the user's kernel
+    connection settings and ssh key passphrase when the user doesn't
+    check the save checkbox.
+    """
+
+    dlg = connection_dialog_factory.submit_filled_dialog(use_keyfile=True,
+                                                         save_settings=False)
+
+    # Press ok and save connection settings
+    qtbot.mouseClick(dlg.accept_btns.button(QDialogButtonBox.Ok),
+                     Qt.LeftButton)
+
+    # create new dialog and check fields
+    new_dlg = connection_dialog_factory.get_default_dialog()
+    assert new_dlg.cf.text() == ""
+    assert not new_dlg.rm_group.isChecked()
+    assert new_dlg.hn.text() == ""
+    assert new_dlg.un.text() == ""
+    assert new_dlg.pn.text() == "22"
+    assert new_dlg.kf.text() == ""
+    if (not sys.platform.startswith('linux') or
+            not os.environ.get('CI') is not None):
+        assert new_dlg.kfp.text() == ""
+
+
+def test_kernel_connection_dialog_remember_input_password(
```

Comment:
remember -> remembers; we at least want to get our function names right since they'll appear in every test report

Reply:
Yup, will do. Haha, you know, I actually agree with all your English suggestions. I'm just slightly annoyed that I missed them :p

Reply:
> I'm just slightly annoyed that I missed them :p

Heh, I've worked as a professional copyeditor and I'm a lot better at English and than I am at programming languages (fortunately...) but if English didn't happen to be my first language, I'm sure I'd find it incredibly confusing and inconsistent. I still miss stuff all the time in my own writing though.

> What do you think about removing the kernel_connection part of the test function names?

I actually replied above to your comment about 10 minutes ago, try reloading the page if it doesn't show up.